### PR TITLE
Feature cleanup

### PR DESCRIPTION
**Pull request by ibgreen**
For #964 

#### Background
- Minor pre-v7 maintenance on the feature subsystem

#### Change List
- Clean up feature system.
- Remove duplicated code.
- Add function support to feature table.
- Add WEBGL2 feature. 
- Remove tree-shaking side effect (explicit table).

### CODE REVIEW

File: modules/core/src/webgl/features/features.js [hunk 1]
```diff
@@ -7,56 +7,57 @@ import WEBGL_FEATURES from './webgl-features-table';
...
+
+  // Get extension name from table
+  const featureDefinition = isWebGL2(gl) ? feature[1] || feature[0] : feature[0];
+
+  let isSupported;
+
+  // Check if the value is dependent on checking one or more extensions
+  if (typeof featureDefinition === 'function') {
+    isSupported = featureDefinition(gl);
+  } else if (Array.isArray(featureDefinition)) {
```
Comment (tsherif) [inline modules/core/src/webgl/features/features.js]: 👍

File: modules/core/src/webgl/features/webgl-features-table.js [hunk 2]
```diff
@@ -1,35 +1,77 @@
...
 export default {
+  [FEATURES.WEBGL2]: [gl => isWebGL2(gl)],
+
   // API SUPPORT
-  VERTEX_ARRAY_OBJECT: ['OES_vertex_array_object', true],
-  TIMER_QUERY: ['EXT_disjoint_timer_query', 'EXT_disjoint_timer_query_webgl2'],
-  INSTANCED_RENDERING: ['ANGLE_instanced_arrays', true],
-  MULTIPLE_RENDER_TARGETS: ['WEBGL_draw_buffers', true],
+  [FEATURES.VERTEX_ARRAY_OBJECT]: ['OES_vertex_array_object', true],
+  [FEATURES.TIMER_QUERY]: ['EXT_disjoint_timer_query', 'EXT_disjoint_timer_query_webgl2'],
```
Comment (tsherif) [inline modules/core/src/webgl/features/webgl-features-table.js]: Make the WebGL 2 version both. Nice use case for your new array functionality.

Comment (ibgreen) [reply modules/core/src/webgl/features/webgl-features-table.js]: > Make the WebGL 2 version both. Nice use case for your new array functionality.

Yes, let's do it separately. For Firefox, so needs separate testing, especially given how this extension has been flipped on and off by browsers.

Comment (ibgreen) [reply modules/core/src/webgl/features/webgl-features-table.js]: You just want to make sure the array code path is actually being used?

Comment (tsherif) [reply modules/core/src/webgl/features/webgl-features-table.js]: Yes. Not a huge deal here, but building out logic that isn't in used anywhere triggers an allergic reaction in me. Can become fertile ground for bit rot and confusion about intention.

**Pull request merged**